packages/itmat-ui-react/src/components/users/manageBatchUsers.tsx

```diff
@@ -0,0 +1,152 @@
+import { Models } from 'itmat-commons';
+import * as React from 'react';
+import { Query, Mutation } from 'react-apollo';
+import { NavLink } from 'react-router-dom';
+import { WHO_AM_I } from 'itmat-commons/dist/graphql/user';
+import css from './userList.module.css';
+import { useQuery } from 'react-apollo';
+import { GET_PROJECT } from 'itmat-commons/dist/graphql/projects';
+import { LoadingBalls } from '../reusable/icons/loadingBalls';
+import { EDIT_USER } from 'itmat-commons/dist/graphql';
+
+export const ManageBatchUsers: React.FunctionComponent = (props) => {
```

Comment:
I am not sure why we need a new page. I'm not clear what it's supposed to do considering there already exists a view for managing a users.

Reply:
seconded

Reply:
As I found that each time the admin could only operate on a single user. Like what I said below about the difference between 'expiredAt' and 'locked'. I think it would be more convenient for the admin to lock a group of users under some conditions, e.g., if he/she want to maintain a project, then the users involved in this project should be all locked temporarily.

Reply:
true, but in that case i would think we should make a user group entity like in *NIX systems to allow managing permissions in group, instead of linking a "group" to a project by filtering projectID.

We should think if the 'user group' concept is really needed though. The only thing you need to edit users in batch is permission. You will not really edit other fields in batch. 
we already have roles attached to each project and each studies. Each role contains users.
Wouldn't it be  easier and more straight forward to implement a "freeze" role function? so if a role is frozen, the user can't perform actions that are permitted under such role. but then they can still perform other stuff if they have other roles.